cql3/cql3_type.cc

```diff
@@ -150,21 +150,36 @@ class cql3_type::raw_collection : public raw {
     virtual cql3_type prepare_internal(const sstring& keyspace, const data_dictionary::user_types_metadata& user_types) override {
         assert(_values); // "Got null values type for a collection";
 
-        if (!is_frozen() && _values->supports_freezing() && !_values->is_frozen()) {
```

Comment:
Commit message has to explain what was the problem with the previous arrangement (if any) and how this patch improves it.